bin/profile/init.js

```diff
@@ -82,6 +91,19 @@ function handleProfileInit(args) {
 
 // Private helper functions
 
+function detectAuthMode(args) {
+    return UserAuthenticator.create(args.url)
+        .then(userAuthenticator => {
+            if (!userAuthenticator) {
```

Comment:
If creation failed, it would result in a rejected promise, and we would never end up here. So `null` response indicates no need for userAuthenticator. It is like Nullable\<T\>.